### PR TITLE
Blip: Deprecate `BlipModel`

### LINKED ISSUE #31199: Some weights of BlipModel were not initialized from the model checkpoint at Salesforce/blip-image-captioning-base.
### System Info

Some weights of BlipModel were not initialized from the model checkpoint at Salesforce/blip-image-captioning-base. This bug is thrown when I run the offical code in https://huggingface.co/docs/transformers/v4.41.3/en/model_doc/blip#transformers.BlipModel
截图 2024-06-03 16-03-57

below is the offical code. 
```python
from PIL import Image
import requests
from transformers import AutoProcessor, BlipModel

model = BlipModel.from_pretrained("Salesforce/blip-image-captioning-base")
processor = AutoProcessor.from_pretrained("Salesforce/blip-image-captioning-base")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

inputs = processor(
    text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True
)

outputs = model(**inputs)
logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
``` 
 @ArthurZucker and @younesbelkada @amyeroberts

### Who can help?

 @ArthurZucker and @younesbelkada @amyeroberts

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

from PIL import Image
import requests
from transformers import AutoProcessor, BlipModel

model = BlipModel.from_pretrained("Salesforce/blip-image-captioning-base")
processor = AutoProcessor.from_pretrained("Salesforce/blip-image-captioning-base")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

inputs = processor(
    text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True
)

outputs = model(**inputs)
logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities

### Expected behavior

The model weight will be initialized correctly when running the official code

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes: https://github.com/huggingface/transformers/issues/31199

As pointed out in https://github.com/huggingface/transformers/issues/28034 and https://github.com/huggingface/transformers/issues/25024 , specifically here: https://github.com/huggingface/transformers/issues/28034#issuecomment-1859735216 `BlipModel` is a copy-pasta from CLIPModel and cannot be used as it is using blip weights on the Hub. We should deprecate this class as it bring a lot of confusion to users and progressively remove it from the docs, then remove the `BlipModel` class itself in the future. 

cc @amyeroberts

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
SGTM! Could you add a deprecation message on the model page blip.md too?
